libraries/Microsoft.Bot.Builder/Bot.cs

```diff
@@ -13,7 +14,9 @@ public class Bot
     {
         private ActivityAdapterBase _adapter;
         private readonly Middleware.MiddlewareSet _middlewareSet = new Middleware.MiddlewareSet();
-        Func<IBotContext, Task> _onReceive = null; 
+        Func<IBotContext, Task> _onReceive = null;
+
+        private static HttpClient _httpClient;
```

Comment:
Consider initializing here, rather than in the property. As this code sits now, the if/null check in the property is subject to race conditions and needs to be changed. The static init rules, if you did that here, would fix that. 
`private static HttpClient _httpClient = new HttpClient()`

With that said, I think sharing a SINGLE HttpClient across all the possible middleware may be a bit too much. Given the non-thread safty around the properties and headers, this may be asking for trouble. (I've just been reading the StackOverflow and other docs on this topic). 

Conceptually I would be good with an HttpClient in each of the relevant spots (QnAMaker, etC). This is a bit more resource consumption, but seems to greatly reduce the risk of side-effects on non-thread safe methods. If we *only* use the QnAMaker one in QnAMaker, then any issues that arise are likley to be easier to debug. 

I am a bit worried about long lived connections and DNS caching. Per [this discussion](https://github.com/dotnet/corefx/issues/11224). I can't tell from the docs if .Net Standard 2 has a ServicePointManager class. To complicate this, services like QnA Maker (and LUIS, and BotFramework) all use AnyCast DNS and endpoints may change at unexpected times. 

The old code (new HttpClient per request) would continually refresh the DNS. The new code won't, as far as I can tell. I think we can live with this, but would love to know if you see any options... 

Why is this stuff never easy? :(

Reply:
By the way, thank you for submitting this. It's hugely helpful!!

Reply:
Sure! I'm enjoying the hack and figured I should contribute. 

I have made considerations to the HttpClient being used with properties such as DefaultHeaders being changed. In my original branch I started on, I converted the DefaultHeaders modifications in Translator.cs to use a HttpRequestMessage. I think this will prevent the issue with thread safety. The other property changed was the Timeout that was set and so I set a CancellationToken to be called with the request instead of setting the property on HttpClient.

https://github.com/RyanDawkins/botbuilder-dotnet/commit/83a51220b5bec644c30c13a2cbb98cd24c3801a3#diff-4f04237a0a45dcb0c61b59dff4624bf4R35